src/magnify/apps/core/permissions.py

```diff
@@ -0,0 +1,24 @@
+"""Permission handlers for Magnify's core app."""
+from rest_framework import permissions
+
+from magnify.apps.core import models
+
+
+class IsObjectAdministrator(permissions.BasePermission):
+    """
+    Allow a request to proceed only if the user is listed as admin to the targeted object.
+    """
+
+    def has_permission(self, request, view):
+        """
+        Allow the request.
+        Only if the playlist exists and the current logged in user is one
+        of its administrator.
+        """
+        if view.action in ["create", "list", "retrieve"]:
```

Comment:
Why filtering the action here ? Why not using this permission in your view only for the action you want ? I don't know if it's the permission responsibility to do this.

Reply:
I would say these checks are indeed related to this permission which applies to all objects with related "administrators".

Reply:
Ouch the checks were wrong and I found another method to check for object permissions. It's all in a fixup.